### DISCUSSION TITLE
Saving Custom Templates

### DISCUSSION THREAD

**Discussion by Niko-k98** [NVIDIA/NVFlare#2720] [Q&A] [answered]
### Python version (`python3 -V`)

3.10.12

### NVFlare version (`python3 -m pip list | grep "nvflare"`)

2.4

### NVFlare branch (if running examples, please use the branch that corresponds to the NVFlare version, `git branch`)

2.4

### Operating system

Red Hat Enterprise Linux (RHEL) 8.3

### Have you successfully run any of the following examples?

- [X] hello-numpy-sag with simulator
- [X] hello-pt with simulator
- [ ] hello-numpy-sag with POC
- [X] hello-pt with POC

### Please describe your question

I am currently running a series of Federated Learning simulations in both simulation and POC. i have been utilizing job templates such as sag_pt sag_pt_he and sag_deploy_map. I have also setup logging with weights and biases. At present every time I make a new job I need to manually add things such as wandb logging to the config scripts. Is there any documentation on how I could go about creating a custom job template? Such that I can run:

 nvflare job create -j jobs/my_job -w  my_custom_template

Where the template would have the necessary content to establish a new job. Thank you for your assistance.

**Comment by YuanTingHsieh** [comment, marked as answer]
Hello @Niko-k98 

Thanks for your question!

You can just put your thing into the following folder structure:

```
my_custom_template/
    config_fed_client.conf
    config_fed_server.conf
    info.conf
    info.md
    meta.json
```

And put your custom code:

```
your_custom_code_folder/
    custom_code1.py
    custom_code2.py
    custom_code3.py
```

You can just copy your job template to where the NVFlare templates are (for example: `"/home/yuantingh/NVFlare/job_templates"`.

```
cp -r my_custom_template /home/yuantingh/NVFlare/job_templates
```

Then you do `nvflare config -jt "/home/yuantingh/NVFlare/job_templates"`

After that you can do:

`nvflare job create -j jobs/my_job -w my_custom_template`
